common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -266,6 +266,21 @@ def get_items(self, course_key, **kwargs):
         return store.get_items(course_key, **kwargs)
 
     @strip_key
+    def get_course_summaries(self, **kwargs):
+        """
+        Returns a list containing the course information in CourseSummary objects.
+        Information contains `location`, `display_name`, `locator` of the courses in this modulestore.
+        """
+        course_summaries = {}
+        for store in self.modulestores:
+            for course_summary in store.get_course_summaries(**kwargs):
+                course_id = self._clean_locator_for_mapping(locator=course_summary.id)
+                if course_id not in course_summaries:
+                    # course is indeed unique. save it in result
```

Comment:
`get_courses` is implemented like this because there's more than one modulestore. This code is at the "mixed" level which sits on top of the old Mongo and Split modulestores. It reads all course IDs from one modulestore, then reads them from the other. Seems like it's accounting for course IDs which may be the same in both modulestores?